exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExporterWorkerThread.java

```diff
@@ -118,15 +127,13 @@ void export() {
 
   @Override
   public void run() {
-    while (true) {
+    while (!Thread.currentThread().isInterrupted()) {
       try {
         export();
-        Thread.sleep(scheduleDelayMillis);
-      } catch (InterruptedException ie) {
-        // Preserve the interruption status as per guidance and stop doing any work.
-        Thread.currentThread().interrupt();
-        return;
+      } catch (Throwable e) {
+        logger.log(Level.WARNING, "Exception thrown by the Stackdriver stats exporter.", e);
       }
+      Uninterruptibles.sleepUninterruptibly(scheduleDelayMillis, TimeUnit.MILLISECONDS);
```

Comment:
Why did you switch to sleepUninterruptibly?  Couldn't you just add another catch block for Throwable?

Reply:
I could but I think this is nicer. Also I think is better to end the Thread nicely.

Reply:
I'm not sure I understand.  The only difference I see is that the new implementation always sleeps for the full time interval, but the old implementation could return early on interruption.  That behavior doesn't matter now, though, since we don't have a shutdown method.   If we added a shutdown method, I think it would be better for the thread to return as soon as possible.

Reply:
Reverted to limit the changes in this PR because we need to cherry pick it.